CMakePresets.json

```diff
@@ -2,19 +2,148 @@
   "version": 3,
   "configurePresets": [
     {
-      "name": "Ninja",
-      "generator": "Ninja Multi-Config",
+      "name": "default",
+      "hidden": true,
+      "generator": "Ninja",
+      "binaryDir": "out/build/${presetName}",
```

Comment:
since clion does a dir per preset in the source folder, we would have way too many, so lets normalize it to vs defaults